src/use_case/data_access_interface/ConversationDataAccessInterface.java

```diff
@@ -2,7 +2,7 @@
 
 import entity.PromptGenerator;
 
-public interface PromptDataAccessInterface {
+public interface ConversationDataAccessInterface {
     PromptGenerator getPromptGenerator();
     // getPromptGenerator in DAO should be pulling the ConversationHistory to then construct a PromptGenerator off of
     void save(PromptGenerator promptGenerator);
```

Comment:
Change this to Conversation History entity. Within other classes, we can just do promptGenerator.getConversationHistory() to get ConversationHistory.